qiskit/primitives/statevector_estimator.py

```diff
@@ -0,0 +1,156 @@
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Estimator class
+"""
+
+from __future__ import annotations
+
+from collections.abc import Iterable
+from typing import Optional, Union
+
+import numpy as np
+from numpy.typing import NDArray
+from pydantic import Field
+from pydantic.types import PositiveInt
+
+from qiskit.quantum_info import SparsePauliOp, Statevector
+
+from .base import BaseEstimatorV2
+from .containers import (
+    BasePrimitiveOptions,
+    BasePrimitiveOptionsLike,
+    EstimatorTask,
+    EstimatorTaskLike,
+    PrimitiveResult,
+    TaskResult,
+    make_data_bin,
+)
+from .containers.dataclasses import mutable_dataclass
+from .primitive_job import PrimitiveJob
+from .utils import bound_circuit_to_instruction
+
+
+@mutable_dataclass
+class ExecutionOptions(BasePrimitiveOptions):
+    """Options for execution."""
+
+    shots: Optional[PositiveInt] = None
+    seed: Optional[Union[int, np.random.Generator]] = None
+
+
+@mutable_dataclass
+class Options(BasePrimitiveOptions):
+    """Options for the primitives.
+
+    Args:
+        execution: Execution time options. See :class:`ExecutionOptions` for all available options.
+    """
+
+    execution: ExecutionOptions = Field(default_factory=ExecutionOptions)
+
+
+class Estimator(BaseEstimatorV2):
+    """
+    Simple implementation of :class:`BaseEstimatorV2` with Statevector.
+
+    :Execution Options:
+
+        - **shots** (None or int) --
+          The number of shots. If None, it calculates the exact expectation
+          values. Otherwise, it samples from normal distributions with standard errors as standard
+          deviations using normal distribution approximation.
+
+        - **seed** (np.random.Generator or int) --
+          Set a fixed seed or generator for the normal distribution. If shots is None,
+          this option is ignored.
+    """
+
+    _options_class = Options
+    options: Options
+
+    def __init__(self, *, options: Optional[BasePrimitiveOptionsLike] = None):
+        """
+        Args:
+            options: Options including shots, seed.
+        """
+        if options is None:
+            options = Options()
+        elif not isinstance(options, BasePrimitiveOptions):
+            options = Options(**options)
+        super().__init__(options=options)
+
+    def run(self, tasks: Iterable[EstimatorTaskLike]) -> PrimitiveJob[PrimitiveResult[TaskResult]]:
+        coerced_tasks = [
+            task if isinstance(task, EstimatorTask) else EstimatorTask.coerce(task)
+            for task in tasks
+        ]
```

Comment:
I don't think `isinstance` check is necessary because it's done in `coerce` method.
```suggestion
        coerced_tasks = [EstimatorTask.coerce(task) for task in tasks]
```

Reply:
Right. Thanks!

Reply:
Sorry. My suggestion is not complete because `Tuple[QuantumCircuit, ObservablesArrayLike, BindingsArrayLike]` is EstimatorTaskLike and Iterable. So, if users give a tuple, it is decomposed into each item by `for task in tasks`.

Reply:
It might be good to extend `coerce` to accept `Iterable[EstimatorTaskLike]` perhaps.

Reply:
Your suggestion is right because users cannot pass single `EstimatorTaskLike` now (I updated the type hint from the review comments).